bubble_sort.rb

```diff
@@ -0,0 +1,41 @@
+# rubocop: disable Metrics/MethodLength
+# frozen_string_literal: true
+
+def bubble_sort(array)
+  array_length = array.size
+  array if array_length <= 1
+  loop do
+    swapped = false
+    (array_length - 1).times do |i|
+      if array[i] > array[i + 1]
+        array[i], array[i + 1] = array[i + 1], array[i]
+        swapped = true
+      end
+    end
+    break unless swapped
+  end
+  array
+end
+
+def bubble_sort_by(array)
+  check = true
+  while check
+    check = false
+    (0...array.length - 1).each do |i|
+      next unless yield(array[i], array[i + 1]).positive?
+
+      array[i], array[i + 1] = array[i + 1], array[i]
+      check = true
+    end
+  end
+  sorted_array
```

Comment:
- [ ] I see you changed the variable from sorted_array to array to reduce line length, but you forgot to change this line too. (It is not recognizing the variable)